synchronization/timer.go

```diff
@@ -0,0 +1,31 @@
+package synchronization
+
+import "time"
+
+type Timer struct {
+	timer *time.Timer
+	C     <-chan time.Time
```

Comment:
Why is C public?

Reply:
It's also public in the original Timer, so I think to maintain the same interface

Reply:
correct, the idea would be that for the buggy operations, you can use the built-in time or our synchronization package without any API changes (or any code changes, just replace the constructing method with ours)

Reply:
So why don't we use the their channel like the documentation days:
```
if !t.Stop() {
	<-t.C
}
t.Reset(d)
```